march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator_controller.py

```diff
@@ -313,3 +337,41 @@ def redo(self):
     def save_changed_joints(self, joints):
         self.joint_changed_history.append(joints)
         self.joint_changed_redo_list = RingBuffer(capacity=100, dtype=list)
+
+    def toggle_lock_startpoint(self, value):
+        if value and self.previous_subgait is not None:
+            rospy.loginfo('Lock that start to ' + self.previous_subgait.subgait_name)
+
+    def toggle_lock_endpoint(self, value):
+        if value and self.previous_subgait is not None:
```

Comment:
Got a big red error when I tried to press this button without any next gait loaded:wink:
```suggestion
        if value and self.next_subgait is not None:
```

Reply:
Oopsie